append_lifecycle.go

```diff
@@ -174,9 +207,61 @@ func NewAppender(ctx context.Context, d Driver, opts *AppendOptions) (*Appender,
 	return a, t.Shutdown, r, nil
 }
 
+type idxAt struct {
+	idx uint64
+	at  time.Time
+}
+
+type integrationStats struct {
+	sample atomic.Pointer[idxAt]
+}
+
+func newIntegrationStats() *integrationStats {
+	return &integrationStats{}
+}
+
+func (i *integrationStats) saw(idx uint64) {
+	i.sample.CompareAndSwap(nil, &idxAt{idx: idx, at: time.Now()})
```

Comment:
This will only work the first time, and then `i.sample` will be non-nil and this will always fail... is that intentional?

Reply:
It gets reset below in the `latency` func - I've added a bunch of comments to make this clearer.